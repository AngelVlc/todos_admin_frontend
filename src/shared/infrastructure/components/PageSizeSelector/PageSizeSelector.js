import React from "react";

export const PageSizeSelector = (props) => {
  const onChange = (event) => {
    props.changePagination({
      ...props.paginationInfo,
      pageSize: event.target.value,
    });
  };

  return (
    <div>
      <span>Per page: </span>
      <select
        onChange={(e) => onChange(e)}
        value={props.paginationInfo.pageSize}
        data-testid='pagesize-select'
      >
        <option value="5">5</option>
        <option value="10">10</option>
        <option value="20">20</option>
      </select>
    </div>
  );
};
