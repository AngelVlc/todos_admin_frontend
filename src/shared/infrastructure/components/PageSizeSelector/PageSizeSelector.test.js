import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import { PageSizeSelector } from "./PageSizeSelector";

describe("PageSizeSelector", () => {
  const mockedChangePagination = jest.fn();

  const renderComponent = (paginationInfo) => {
    return {
      ...render(
        <PageSizeSelector
          paginationInfo={paginationInfo}
          changePagination={mockedChangePagination}
        />
      ),
    };
  };

  afterEach(cleanup);

  it("Should match the snapshot", () => {
    const { asFragment } = renderComponent({
      pageSize: "10",
    });

    expect(asFragment(<PageSizeSelector />)).toMatchSnapshot();
  });

  it("Should change the page size", async () => {
    const { getByTestId } = renderComponent({
      pageSize: 10,
    });

    await waitFor(() => {
      fireEvent.change(getByTestId("pagesize-select"), {
        target: { value: 20 },
      });
    });

    expect(mockedChangePagination.mock.calls.length).toBe(1);
    expect(mockedChangePagination.mock.calls[0][0]).toStrictEqual({
      pageSize: "20",
    });
  });
});
