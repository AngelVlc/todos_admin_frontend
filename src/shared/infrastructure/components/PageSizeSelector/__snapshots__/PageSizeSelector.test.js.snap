// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSizeSelector Should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <span>
      Per page: 
    </span>
    <select
      data-testid="pagesize-select"
    >
      <option
        value="5"
      >
        5
      </option>
      <option
        value="10"
      >
        10
      </option>
      <option
        value="20"
      >
        20
      </option>
    </select>
  </div>
</DocumentFragment>
`;
